integrationtest/org/daisy/dotify/formatter/test/LayoutEngineTest.java

```diff
@@ -156,24 +158,21 @@ public void testLayoutEngineContentItemsNoFallback() throws
         );
     }
 
-    public int compareBinary(InputStream f1, InputStream f2) throws IOException {
-        InputStream bf1 = new BufferedInputStream(f1);
-        InputStream bf2 = new BufferedInputStream(f2);
-        int pos = 0;
-        try {
-            int b1;
-            int b2;
-            while ((b1 = bf1.read()) != -1 & b1 == (b2 = bf2.read())) {
-                pos++;
-                // continue
-            }
-            if (b1 != -1 || b2 != -1) {
-                return pos;
+    public int compareText(InputStream f1, InputStream f2) throws IOException {
+        BufferedReader br1 = new BufferedReader(new InputStreamReader(f1));
+        BufferedReader br2 = new BufferedReader(new InputStreamReader(f2));
+
+        int linePos = 1;
+        String line;
+        while((line = br1.readLine()) != null) {
+            if(!line.equals(br2.readLine())) {
+                return linePos;
```

Comment:
You return here without closing `br1` and `br2`.

Reply:
Fixed, great catch. Thank you.

Reply:
It would be nicer to make use of a `try..catch` construction. Now you have a piece of code which is duplicated, also the method has two exit points (which I wouldn't prefer).